src/plugins.js

```diff
@@ -0,0 +1,606 @@
+ * @param {html} html - The object for handling post pug processing (raw HTML string)
+ */
+async function _registerHTML(plugin, html) {
+    return new Promise((resolve, reject) => {
+        if (!plugin || typeof plugin !== 'string') {
+            reject(new NoNameError('No plugin name provided'))
+        }
+
+        if (!html || typeof html !== 'function') {
+            reject(new NoOptionError('No handler provided'))
+        }
+
+        if (_inList(plugin, htmls)) {
+            reject(new PluginExistsError('HTML handler already registered'))
+        }
+
+        htmls.push({
+            plugin: plugin,
+            handler: html
+        })
+
+        _addPlugin(plugin, 'html', htmls.length-1)
+
+        resolve(true)
+    })
+}
+
+/**
+ * Register a first pass page handler (generate place holder content)
+ * @param {pageFirst} first - The object for manipulating page, first pass
+ */
+async function _registerPageFirst(plugin, first) {
+    return new Promise((resolve, reject) => {
+        if (!plugin || typeof plugin !== 'string') {
+            reject(new NoNameError('No plugin name provided'))
+        }
+
+        if (!first || typeof first !== 'function') {
+            reject(new NoOptionError('No handler provided'))
+        }
+
+        if (_inList(plugin, pageFirsts)) {
+            reject(new PluginExistsError('Page first pass handler already registered'))
+        }
+
+        pageFirsts.push({
+            plugin: plugin,
+            handler: first
+        })
+
+        _addPlugin(plugin, 'pageFirst', pageFirsts.length-1)
+
+        resolve(true)
+    })
+}
+
+/**
+ * Register a first pass page handler (generate place holder content)
+ * @param {pageFirst} second - The object for manipulating page, second pass
+ */
+async function _registerPageSecond(plugin, second) {
+    return new Promise((resolve, reject) => {
+        if (!plugin || typeof plugin !== 'string') {
+            reject(new NoNameError('No plugin name provided'))
+        }
+
+        if (!second || typeof second !== 'function') {
+            reject(new NoOptionError('No handler provided'))
+        }
+
+        if (_inList(plugin, pageSeconds)) {
+            reject(new PluginExistsError('Page second pass handler already registered'))
+        }
+
+        pageSeconds.push({
+            plugin: plugin,
+            handler: second
+        })
+
+        _addPlugin(plugin, 'pageSecond', pageSeconds.length-1)
+
+        resolve(true)
+    })
+}
+
+
+/*
+ * ========================================================
+ *                Plugin gets
+ * ========================================================
+ */
+function _getMixins() { return mixins }
+
+function _getWatchers() { return watchers }
+
+function _getPugs() { return pugs }
+
+function _getHTMLs() { return htmls }
+
+function _getPageFirsts() { return pageFirsts }
+
+function _getPageSeconds() { return pageSeconds }
+
+const public = {
```

Comment:
The public object, which is the plugin object to a relaxed-plugin, is provided to the plugin. Since the plugin.js handles the loading of the plugins via:

```javascript
var plug = require('relaxed-example-plugin')
plug.activate(public)
```

It gives a limited API to the plugin to add functionality with.